repository.go

```diff
@@ -546,49 +548,25 @@ func cleanUpDir(path string, all bool) error {
 
 // Config return the repository config. In a filesystem backed repository this
 // means read the `.git/config`.
-func (r *Repository) Config() (*config.Config, error) {
-	return r.Storer.Config()
+func (r *Repository) Config(o ...config.WithOptions) (*config.Config, error) {
```

Comment:
I generally find variadic functions quite neat, and probably something we should look into using more on V6. However, it is important to note that this can break backwards compatibility on this specific case.

For example, if we were to go ahead with this change I believe we would break [argocd-autopilot](https://github.com/argoproj-labs/argocd-autopilot/blob/55eb5621fbd4650604a151b3d0efc2592f7dcfdc/pkg/git/gogit/mocks/repository.go#L133) as they define an [interface](https://github.com/argoproj-labs/argocd-autopilot/blob/55eb5621fbd4650604a151b3d0efc2592f7dcfdc/pkg/git/gogit/repo.go#L22) based on our previous signature.